nanoFramework.System.Net.Http/Http/ByteArrayContent.cs

```diff
@@ -0,0 +1,107 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// Portions Copyright (c) Microsoft Corporation.  All rights reserved.
+// See LICENSE file in the project root for full license information.
+//
+
+using System.IO;
+
+namespace System.Net.Http
+{
+    /// <summary>
+    /// Provides HTTP content based on a byte array.
+    /// </summary>
+    public class ByteArrayContent : HttpContent
+    {
+        private readonly byte[] _content;
+        private readonly int _offset;
+        private readonly int _count;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ByteArrayContent"/> class.
+        /// </summary>
+        /// <param name="content">The content used to initialize the <see cref="ByteArrayContent"/>.</param>
+        /// <exception cref="ArgumentNullException">The <paramref name="content"/> parameter is <see langword="null"/>.</exception>
+        public ByteArrayContent(byte[] content)
+        {
```

Comment:
why not calling directly the other constructor?

Reply:
Need to keep this aligned with the .NET API...

Reply:
why won't `: this(content, 0, content.Length);` won't be aligned with the .NET API? So you just call the other constructor instead of having almost the same code in this one?

Reply:
If `content` is null using it for the param `content.Length` will throw NullRef exception